plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/AbstractRefactoring.java

```diff
@@ -129,13 +121,32 @@ protected void executeCommand(AbstractTransactionalCommand refactoringCommand) {
 	protected boolean internalDoUndo() {
 		return true;
 	}
+	
+	/**
+	 * Combines the internal command execution call with the parsing and linking of the specification elements.
+	 */
+	protected void doInternalExecute() {
+		internalExecute();
+		parseAndLinkSpecificationElements();
+	}
```

Comment:
Two things:
1) Usually "doX..." Methods are sub methods of "X.." methods, means that "X" is calling "doX" internally, here it is the other way around. If we change that, we need to adjust all subclasses to implement the "doInternalExecute" method instead of "internalExecute".

2) Now that refactoring execution is combined with parse and link functionality we can remove these calls from the tests!